news/utils.py

```diff
@@ -0,0 +1,138 @@
+import re
+import requests
+from bs4 import BeautifulSoup, NavigableString
+from usp.tree import sitemap_tree_for_homepage
+
+def build_url_index(base_url, keyword=None):
+
+    if keyword is None:
+        keyword = [""]
+    elif isinstance(keyword, str):
+        keyword = [keyword]
+
+    tree = sitemap_tree_for_homepage(base_url)
+    page_index = [page.url for key in keyword for idx, page in enumerate(tree.all_pages()) if key in page.url]
+    return page_index
+
+
+def get_text_from_page(url=None, html_path=None, tag="article", attrs=None):
+
+    assert bool(url is not None) != bool(html_path is not None)
+
+    if url is not None:
```

Comment:
I think it would be a lot simpler if this function was just html -> text instead of having the ability to pull the page.

Thoughts on if you should pass in raw html strings or the Soup? The former makes it easier to use, the ladder makes it easier to customize the parsing. Unsure what is better.